crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1262,16 +1262,64 @@ where
                 is_async,
                 ..
             }) => {
-                for item in items {
-                    self.visit_expr(&item.context_expr);
-                    if let Some(optional_vars) = item.optional_vars.as_deref() {
-                        self.add_standalone_expression(&item.context_expr);
-                        self.push_assignment(CurrentAssignment::WithItem {
-                            item,
-                            is_async: *is_async,
-                        });
+                for item @ ruff_python_ast::WithItem {
+                    range: _,
+                    context_expr,
+                    optional_vars,
+                } in items
+                {
+                    let context_manager = self.add_standalone_expression(context_expr);
```

Comment:
This line looks redundant with line 1274? We shouldn't add `context_expr` as a standalone expression _twice_, and we shouldn't need to do it at all if there are no `optional_vars` (meaning we aren't going to create any Definition or register any attribute assignment).